test/e2e/framework/test_context.go

```diff
@@ -426,6 +426,7 @@ func RegisterClusterFlags(flags *flag.FlagSet) {
 	flags.DurationVar(&TestContext.timeouts.SystemPodsStartup, "system-pods-startup-timeout", TestContext.timeouts.SystemPodsStartup, "Timeout for waiting for all system pods to be running before starting tests.")
 	flags.DurationVar(&TestContext.timeouts.NodeSchedulable, "node-schedulable-timeout", TestContext.timeouts.NodeSchedulable, "Timeout for waiting for all nodes to be schedulable.")
 	flags.DurationVar(&TestContext.timeouts.SystemDaemonsetStartup, "system-daemonsets-startup-timeout", TestContext.timeouts.SystemDaemonsetStartup, "Timeout for waiting for all system daemonsets to be ready.")
+	flags.DurationVar(&TestContext.timeouts.NetPolSleepBeforeProbing, "netpol-sleep-before-probing", TestContext.timeouts.NetPolSleepBeforeProbing, "Time to wait before the first attempt probing Pod connectivity when validating a NetworkPolicy.")
```

Comment:
I'm not in favor of adding flags to special cast specific tests , furthermore, the e2e test should represent the expected behavior and should not be customized to have different behaviors depending on where it runs on how it is configured.

In this case, is ok that users deploy a network policy and take 1 minute more than in other deployments?

The e2e test should standardize the behavior

Reply:
Fully agreed. Test-specific flags don't belong into the framework:
https://github.com/kubernetes/kubernetes/blob/5961d8ce1bb4eb6537f7ca3b3eaf7fd9d5cd2595/test/e2e/framework/test_context.go#L57-L84

`framework/config` might not work as advertised anymore. Viper also is gone? I need to revisit this part.

But what works is adding flags in your tests.